cmd/bundle/deploy.go

```diff
@@ -14,20 +16,30 @@ var deployCmd = &cobra.Command{
 	RunE: func(cmd *cobra.Command, args []string) error {
 		b := bundle.Get(cmd.Context())
 
-		// If `--force` is specified, force acquisition of the deployment lock.
-		b.Config.Bundle.Lock.Force = forceDeploy
-
-		return bundle.Apply(cmd.Context(), b, bundle.Seq(
-			phases.Initialize(),
-			phases.Build(),
-			phases.Deploy(),
-		))
+		return deploy(cmd, b)
 	},
 }
 
+func deploy(cmd *cobra.Command, b *bundle.Bundle) error {
+	// If `--force` is specified, force acquisition of the deployment lock.
+	b.Config.Bundle.Lock.Force = forceDeploy
+
+	if computeID == "" {
+		computeID = os.Getenv("DATABRICKS_CLUSTER_ID")
```

Comment:
The field should also be called `cluster_id` if we pull the cluster ID. We can have one for each of clusters, compute, warehouses. The flag parsing + environment overrides should happen centrally and not by reusing globals from other commands.

Reply:
compute and compute_id are the general terms we're using for serverless jobs / interactive to indicate both clusters and "serverless" clusters. Warehouses are separate. I'd rather not change the name to cluster_id if it's really an id that also works for serverless compute.

Reply:
OK, `computeID` works. Can you remove the getenv check though? It will mean that deployment fails if that value is set if it's not in dev mode, which would be unexpected behavior to me. Having the flag alone captures the intent without that failure mode.

Reply:
I updated it so we only fail if the override is set in the bundle or CLI, and not when it's set in the environment.